driver.go

```diff
@@ -159,8 +165,8 @@ func connect(opts *Options) (*Conn, error) {
 
 	httpTransport, ok := transport.(*thrift.THttpClient)
 	if ok {
-		// TODO: currently masking as a python connector until additional user agents are white listed.
-		httpTransport.SetHeader("User-Agent", "godatabrickssqlconnector/0.9.0 (go-dbsql)")
+		userAgent := fmt.Sprintf("godatabrickssqlconnector/0.9.0 (%s)", opts.ClientId)
```

Comment:
It would be helpful to capture other information as well (Platform, Go lang version).

see this as an example how this is done in one of the azure sdks:
https://github.com/Azure/azure-cosmosdb-node/blob/ef4b113757b72356475f6bad994dc87f17e49291/source/lib/platform.js#L43-L57

Reply:
I know that the user agent can be messy (e.g. Chrome's user agent can be "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/536.5 (KHTML, like Gecko) Chrome/19.0.1084.52 Safari/536.5") but wondering what format would make the most sense here. Following the example you provided, does something like `{os name}/{os version} Golang/{golang version} godatabrickssqlconnector/{connector version} ` work?

Reply:
thanks @mattdeekay let's do this outside of the scope of this PR.